src/main/java/com/epam/processor/DataProcessor.java

```diff
@@ -23,101 +23,196 @@ public DataProcessor(List<RoadAccident> roadAccidentList){
+
+        while (roadAccidentIterator.hasNext()) {
+            RoadAccident current = roadAccidentIterator.next();
+            result.put(current.getDistrictAuthority(), current.getAccidentId());
+        }
+
+        return result;
     }
 
 
     // Now let's do same tasks but now with streaming api
 
 
+    public RoadAccident getAccidentByIndex(String index) {
+        Optional<RoadAccident> result = roadAccidentList.stream()
+                .filter(acc -> acc.getAccidentId().equals(index))
+                .findFirst();
 
-    public RoadAccident getAccidentByIndex(String index){
-        return null;
+        return result.isPresent() ? result.get() : null;
     }
 
 
     /**
      * filter list by longtitude and latitude fields
+     *
      * @param minLongitude
      * @param maxLongitude
      * @param minLatitude
      * @param maxLatitude
      * @return
      */
-    public Collection<RoadAccident> getAccidentsByLocation(float minLongitude, float maxLongitude, float minLatitude, float maxLatitude){
-        return null;
+    public Collection<RoadAccident> getAccidentsByLocation(float minLongitude, float maxLongitude, float minLatitude, float maxLatitude) {
+        return roadAccidentList.stream()
+                .filter(acc -> minLongitude <= acc.getLongitude() && acc.getLongitude() <= maxLongitude &&
+                        minLatitude <= acc.getLatitude() && acc.getLatitude() <= maxLatitude)
+                .collect(Collectors.toList());
     }
 
     /**
      * find the weather conditions which caused max number of incidents
+     *
      * @return
      */
-    public List<String> getTopThreeWeatherCondition(){
-        return null;
+    public List<String> getTopThreeWeatherCondition() {
+
+        Map<String, Long> mapAccidentsByWeatherCondition = roadAccidentList.stream()
+                .collect(Collectors.groupingBy(RoadAccident::getWeatherConditions, Collectors.counting()));
+
+        return mapAccidentsByWeatherCondition.entrySet().stream()
+                .sorted((s1, s2) -> s2.getValue().compareTo(s1.getValue()))
+                .map(Map.Entry::getKey)
+                .limit(3)
+                .collect(Collectors.toList());
     }
 
     /**
      * count incidents by road surface conditions
+     *
      * @return
      */
-    public Map<String, Long> getCountByRoadSurfaceCondition(){
-        return null;
+    public Map<String, Long> getCountByRoadSurfaceCondition() {
+        return roadAccidentList.stream()
+                .collect(Collectors.groupingBy(RoadAccident::getRoadSurfaceConditions, Collectors.counting()));
     }
 
     /**
      * To match streaming operations result, return type is a java collection instead of multimap
+     *
      * @return
      */
-    public Map<String, List<String>> getAccidentIdsGroupedByAuthority(){
-        return null;
+    public Map<String, List<String>> getAccidentIdsGroupedByAuthority() {
+
+        return roadAccidentList.stream()
```

Comment:
You don't really need to specify a HashMap::new here - it will work fine without it.